kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemViewKeyInputFormat.java

```diff
@@ -76,24 +77,27 @@ public FileSystemViewKeyInputFormat(FileSystemDataset<E> dataset,
     Format format = dataset.getDescriptor().getFormat();
 
     boolean isSpecific = SpecificRecord.class.isAssignableFrom(dataset.getType());
+    String readerSchema = conf.get("kite.readerSchema");
+    if (isSpecific && readerSchema == null) {
```

Comment:
To ensure the reader schema is consistent across all files, we should determine what it is before running the job. But it always needs to be based on the record that will hold the contents. For specific, it should be the specific record's schema and for reflect (reading) it should be the AllowNull schema of the requested type. For generic, it should default to the dataset schema, but can be overridden with a projected schema or a less restrictive schema.